data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters.py

```diff
@@ -1,131 +1,264 @@
+OR 
+    (observation_concept_id = 1585873 AND (value_as_number < 0 OR value_as_number > 99))
+OR
+    (observation_concept_id = 1586159 AND (value_as_number < 0 OR value_as_number > 99))
+OR
+    (observation_concept_id = 1586162 AND (value_as_number < 0 OR value_as_number > 99)))
+""")
 
-UNION ALL
-
-SELECT
-  *
+CLEAN_PPI_NUMERIC_FIELDS_QUERY = jinja_env.from_string("""
+SELECT 
+    observation_id,
+    person_id,
+    observation_concept_id,
+    observation_date,
+    observation_datetime,
+    observation_type_concept_id,
+CASE
+    WHEN observation_concept_id = 1585889 AND (value_as_number < 0 OR value_as_number > 20)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1585890 AND (value_as_number < 0 OR value_as_number > 20)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1585795 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1585802 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1585820 AND (value_as_number < 0 OR value_as_number > 255)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1585864 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1585870 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1585873 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN NULL
+    
+    WHEN observation_concept_id = 1586159 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN NULL
+       
+    WHEN observation_concept_id = 1586162 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN NULL
+    
+    ELSE value_as_number
+END AS
+    value_as_number,
+    value_as_string,
+CASE
+    WHEN observation_concept_id = 1585889 AND (value_as_number < 0 OR value_as_number > 20)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585890 AND (value_as_number < 0 OR value_as_number > 20)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585795 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585802 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585820 AND (value_as_number < 0 OR value_as_number > 255)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585864 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585870 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1585873 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1586159 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
+    
+    WHEN observation_concept_id = 1586162 AND (value_as_number < 0 OR value_as_number > 99)
+    THEN 2000000010
```

Comment:
Suggestion: Query is processing slightly quicker when I restructured the case statement like this.

`CASE
    WHEN observation_concept_id IN (1585889, 1585890) AND (value_as_number < 0 OR value_as_number > 20) THEN NULL
    WHEN observation_concept_id IN (1585795, 1585802, 1585864, 1585870, 1585873, 1586159, 1586162) AND (value_as_number < 0 OR value_as_number > 99) THEN NULL
    WHEN observation_concept_id = 1585820 AND (value_as_number < 0 OR value_as_number > 255) THEN NULL
  ELSE`

Reply:
Modified both `CASE` statements in `CLEAN_PPI_NUMERIC_FIELDS_QUERY`